### PR TITLE
Refactor sharpness

### PR DESCRIPTION
**Pull request by aaronmondal**
Part of #2279.

Changes:
1. Made `sharpness_image` private (`_sharpness_image` now), as `sharpness` already handles single images and `_equalize_image` is private as well.
2. Made `sharpness` compatible with grayscale images (previously only worked with RGB).
3. Added the `name` argument and applied a `tf.name_scope`.
4. Decorated with `@tf.function`. **Now runs ~60% faster.** ([colab](https://colab.research.google.com/drive/1uZzz4G-Bfme1-9jaBsMPYTzykA25GsTu#scrollTo=mFA_HXxMmvhd))
5. Removed the `clip_by_value` invocation since the sharpening kernel is already normalized.
6. Refactored the tests. `sharpness` and `equalize` now both handle all input shapes.

Also removed a faulty test for `equalize`. That test was for the now unsupported `channels_first` and was passing because it was just renaming the `name_scope` to `channels_first` :D.

### CODE REVIEW

**Review by WindQAQ — APPROVED** [review]

**Comment by WindQAQ** [comment]
@aaronmondal can you make the colab link public to "anyone who knows the link can access it"? Other than this, the PR LGTM.

**Comment by aaronmondal** [comment]
Link: https://colab.research.google.com/drive/1uZzz4G-Bfme1-9jaBsMPYTzykA25GsTu?usp=sharing
Sorry I always forget that the default link is not public :O

File: tensorflow_addons/image/color_ops.py [hunk 1]
```diff
@@ -79,51 +79,56 @@ def equalize(image: TensorLike, name: Optional[str] = None) -> tf.Tensor:
...
-    kernel = tf.tile(kernel, [1, 1, 3, 1])
-    strides = [1, 1, 1, 1]
+    kernel = tf.tile(kernel, [1, 1, image_channels, 1])
+
+    # Apply kernel channel-wise.
     degenerate = tf.nn.depthwise_conv2d(
-        image, kernel, strides, padding="VALID", dilations=[1, 1]
+        image, kernel, strides=[1, 1, 1, 1], padding="VALID", dilations=[1, 1]
     )
-    degenerate = tf.clip_by_value(degenerate, 0.0, 255.0)
```
Comment (WindQAQ) [inline tensorflow_addons/image/color_ops.py:100]: I'd say we have to keep the clip here because we cann't know if users pass a valid image into the function.

Comment (WindQAQ) [reply tensorflow_addons/image/color_ops.py:100]: NVM. I think you're right.

Comment (aaronmondal) [reply tensorflow_addons/image/color_ops.py:100]: @WindQAQ Ah I see this is inconsistent with `blend`. My thought was if someone were to use an image scaled to `(-1, 1)` or `[-127.5, 127.5]`, this would cut off the bottom half and removing it would enable these scaled images as well. It should work anyways right now, because `blend` clips to `[0, 255]`, but it may make sense to remove the clip in blend to enable images in more scaling ranges. Gotta see whether this works with other ops that make use of `blend` though.

Comment (aaronmondal) [reply tensorflow_addons/image/color_ops.py:100]: Edit: Things still works with the "supported" image ranges, but zero-centered ranges will be cut off by `blend`.

Comment (WindQAQ) [reply tensorflow_addons/image/color_ops.py:100]: I think the clip should be removed because people might want to apply these ops during post-processing. In this case, those images are not supposed to be VALID IMO. Got to check if CV2/scipy have similar functionality.
